sdk/typescript/src/types/common.ts

```diff
@@ -134,7 +140,8 @@ export function generateTransactionDigest(
   const senderSignedDataBytes = bcs
     .ser('SenderSignedData', senderSignedData)
     .toBytes();
-  return sha256Hash('SenderSignedData', senderSignedDataBytes);
+
+  return new Base58DataBuffer(sha256Hash('SenderSignedData', senderSignedDataBytes)).toString();
```

Comment:
For backward compatibility, I think you can pass in a `serializationFmt: 'base58' | 'base64' ` to the function signature of `generateTransactionDigest`. And in line 144, you can do `serializationFmt === 'base58' ? ... : ...)

Reply:
revised